nighres/brain/dots_segmentation.py

```diff
@@ -0,0 +1,719 @@
+                                                           m]])) + \
+                                               0.5 * (s_O_x_m[i,j,k] *
+                                               np.nanmax([prev_iter_U[
+                                                          x_m_s_O[i,j,k,0],
+                                                          x_m_s_O[i,j,k,1],
+                                                          x_m_s_O[i,j,k,2],
+                                                          idx],
+                                                          prev_iter_U[
+                                                          x_m_s_O[i,j,k,0],
+                                                          x_m_s_O[i,j,k,1],
+                                                          x_m_s_O[i,j,k,2],
+                                                          l],
+                                                          prev_iter_U[
+                                                          x_m_s_O[i,j,k,0],
+                                                          x_m_s_O[i,j,k,1],
+                                                          x_m_s_O[i,j,k,2],
+                                                          m]]))
+    return curr_U
+
+
+def _calc_segmentation(U):
+    # Return hard segmentation based on MRF energy U  
+    U_temp = np.copy(U)
+    U_temp[np.isnan(U_temp)] = -np.inf
+    segmentation = np.argmax(U_temp, axis = 3)
+    return segmentation
+
+
+def calc_posterior_probability(l, U, wm_atlas, g0 = None):
+    # Return posterior probability of tract l from MRF energy U (Eq 15)               
+    if wm_atlas == 1:
+        N_t = 23
+        tract_pair_array = np.array([list(i) for i in tract_pair_sets_1])
+    elif wm_atlas == 2:
+        N_t = 41
+        tract_pair_array = np.array([list(i) for i in tract_pair_sets_2])
+    if g0 == None:
+        g0 = N_t
+    idx = np.concatenate(([l], np.where(np.any(tract_pair_array 
+                                               == l, axis=1))[0]+N_t))
+    posterior_l = (np.nansum(np.exp(g0*U[:,:,:,idx]), axis=3) /
+                   np.nansum(np.exp(g0*U),axis=3))
+    return posterior_l
+    
+
+def dots_segmentation(tensor_image, mask, atlas_dir, wm_atlas = 1, 
+                      max_iter = 25, convergence_threshold = 0.005, s_I = 1/42, 
+                      c_O = 0.5, max_angle = 67.5, save_data = False, 
+                      overwrite = False, output_dir = None, file_name = None):
+    """DOTS segmentation
+
+    Segment major white matter tracts in diffusion tensor images using Diffusion
+    Oriented Tract Segmentation (DOTS) algorithm.
+    
+    Parameters
+    ----------
+    tensor_image: niimg
+        Input image containing the diffusion tensor coefficients in the
+        following order: volumes 0-5: D11, D22, D33, D12, D13, D23
+    mask: niimg
+        Binary brain mask image which limits computation to the defined volume.
+   atlas_dir: str, optional
```

Comment:
space before `atlas_dir` one space off, misalignment in built docs

Reply:
Also, this does not seem to be an optional input as there is no default value. So the function will fail if this is not set. Explanation is also not exactly clear